### PR TITLE
Init repo

### PR DESCRIPTION
**Pull request by JamesEarle**

### CODE REVIEW

**Comment by RyanCross** [comment]
I have some addresses that I believe are for the v2 StakingController, not sure if they are still being used but they have some staking request placed transactions on them
mainnet: `0x45b13d8e6579d5C3FeC14bB9998A3640CD4F008D`
rinkeby: `0x7FDd24f30fB8a3E0021e85Fdb737a3483D3C8135`

File: package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "contract-addresses",
```
Comment (Remscar) [inline package.json:2]: The name should be `@zero-tech/zer0-contracts`

File: package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "version": "0.0.1",
+  "description": "A place to store all the contract addresses for the Zero ecosystem",
+  "main": "index.ts",
+  "scripts": {
+    "build": "tsc"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/zer0-os/contract-addresses.git"
```
Comment (Remscar) [inline package.json:11]: Can we rename this GitHub repo to `zer0-contracts` please, and update these references.

File: src/types.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+interface Categories {
+  infra: Addresses;
+  tokens: Addresses;
+  staking: Addresses;
+  sales: Sales;
+}
+
+interface Addresses {
+  [contractName: string]: string;
```
Comment (Remscar) [inline src/types.ts:13]: This needs to be a `Maybe<string>`

File: src/types.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+}
+interface Sales {
+  [saleName: string]: string | SaleSeason;
```
Comment (Remscar) [inline src/types.ts:16]: This needs to be a `Maybe<string | SaleSeason>`

File: src/types.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+interface SaleSeason {
+  [seasonName: string]: string;
```
Comment (Remscar) [inline src/types.ts:20]: This needs to be a `Maybe<string>`

File: src/types.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+export interface Networks {
+  [networkName: string]: Categories;
```
Comment (Remscar) [inline src/types.ts:2]: This needs to be a `Maybe<Categories>`

File: tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (Remscar) [inline tsconfig.json:1]: Use this `tsconfig.json` file:
```
{
  "compilerOptions": {
    "declaration": true,
    "declarationMap": true,
    "lib": [
      "ES2018"
    ],
    "module": "CommonJS",
    "moduleResolution": "node",
    "esModuleInterop": true,
    "strict": true,
    "target": "ES2018",
    "sourceMap": true,
    "composite": true,
    "outDir": "lib",
    "rootDir": "src",
    "strictNullChecks": true
  },
  "include": [
    "src"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

File: package.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  "author": "",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/zer0-os/contract-addresses/issues"
+  },
+  "homepage": "https://github.com/zer0-os/contract-addresses#readme",
+  "devDependencies": {
+    "typescript": "^4.7.2"
+  }
```
Comment (Remscar) [inline package.json:21]: Needs
```
"files": [
    "lib/",
    "src/"
  ],
  ```
  added

File: package.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "contract-addresses",
+  "version": "0.0.1",
+  "description": "A place to store all the contract addresses for the Zero ecosystem",
+  "main": "index.ts",
+  "scripts": {
+    "build": "tsc"
```
Comment (Remscar) [inline package.json:7]: `"build": "tsc --build --verbose ./tsconfig.build.json",`

File: .npmignore [hunk 10]
```diff
@@ -0,0 +1,2 @@
+.github/
+README.md
```
Comment (Remscar) [inline .npmignore:2]: Also add `.circleci`
`workflows`
`tsconfig.json`
`tsconfig.build.json`
`tsconfig.build.tsbuildinfo`

File: .gitignore [hunk 11]
```diff
@@ -6,6 +6,9 @@ yarn-debug.log*
 yarn-error.log*
 lerna-debug.log*
 
+# TypeScript output folder
+lib
+
```
Comment (Remscar) [inline .gitignore:11]: Add
`node_modules`
`*.tsbuildinfo`

File: src/index.ts [hunk 12]
```diff
@@ -0,0 +1,81 @@
+import { Networks } from "./types";
+
+export const networkAddresses: Networks = {
```
Comment (Remscar) [inline src/index.ts:3]: Consider renaming `zer0ProtocolAddresses` or `zeroProtocolAddresses`

File: src/index.ts [hunk 13]
```diff
@@ -0,0 +1,81 @@
+import { Networks } from "./types";
+
```
Comment (Remscar) [inline src/index.ts:2]: consider adding `export * from "./types"`

File: src/types.ts [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+interface Categories {
```
Comment (Remscar) [inline src/types.ts:5]: Might be useful for these to be exported

File: src/types.ts [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+  infra: Addresses;
```
Comment (Remscar) [inline src/types.ts:6]: Maybe it would be better if we split `infra` into `zns` `zauction`. We've already split out staking

**Review by Remscar — COMMENTED** [review]
Maybe overkill, but having linter + prettier in this repo would be nice.

**Comment by Remscar** [comment]
> I have some addresses that I believe are for the v2 StakingController, not sure if they are still being used but they have some staking request placed transactions on them mainnet: `0x45b13d8e6579d5C3FeC14bB9998A3640CD4F008D` rinkeby: `0x7FDd24f30fB8a3E0021e85Fdb737a3483D3C8135`

The zNS Staking Controller isn't going to come back. So we can probably just leave it out

File: .gitignore (hunk 11 above)
Comment (JamesEarle) [reply .gitignore:11]: `node_modules` is already there on line 44
`*.tsbuildinfo` is already there on line 51

**Pull request merged**